File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ret = 0;
        int l = 0;
        int r = 1;

        while (r < prices.size()) {
            if (prices[l] > prices[r]) {
                r += 1;
                l = r - 1;
            }
            else {
                ret = max(ret, prices[r] - prices[l]);
                r += 1;
            }  
        }
        return ret;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        # naive solution w/ division
        product = 1
        numZeros = 0
        ret = []

        for i in range(len(nums)):

            if nums[i] == 0:
                numZeros += 1
                continue
            
            product *= nums[i]

        if numZeros > 1:
            l = [0] * len(nums)
            return l

        for i in range(len(nums)):

            if numZeros > 0:
                if nums[i] == 0:
                    ret.append(product)

                else:
                    ret.append(0)

            else:
                ret.append(int(product / nums[i]))

        return ret
            
